src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php

```diff
@@ -98,33 +99,36 @@ public static function castTraceStub(TraceStub $trace, array $a, Stub $stub, $is
             return array();
         }
         $lastCall = isset($frames[$i]['function']) ? ' ==> '.(isset($frames[$i]['class']) ? $frames[0]['class'].$frames[$i]['type'] : '').$frames[$i]['function'].'()' : '';
+        $frames[] = array('function' => '');
 
         for ($j += $trace->numberingOffset - $i++; isset($frames[$i]); ++$i, --$j) {
-            $call = isset($frames[$i]['function']) ? (isset($frames[$i]['class']) ? $frames[$i]['class'].$frames[$i]['type'] : '').$frames[$i]['function'].'()' : '???';
+            $f = $frames[$i];
+            $call = isset($f['function']) ? (isset($f['class']) ? $f['class'].$f['type'] : '').$f['function'].'()' : '???';
 
-            $a[Caster::PREFIX_VIRTUAL.$j.'. '.$call.$lastCall] = new FrameStub(
+            $label = $call.$lastCall;
+            $frame = new FrameStub(
                 array(
-                    'object' => isset($frames[$i]['object']) ? $frames[$i]['object'] : null,
-                    'class' => isset($frames[$i]['class']) ? $frames[$i]['class'] : null,
-                    'type' => isset($frames[$i]['type']) ? $frames[$i]['type'] : null,
-                    'function' => isset($frames[$i]['function']) ? $frames[$i]['function'] : null,
+                    'object' => isset($f['object']) ? $f['object'] : null,
+                    'class' => isset($f['class']) ? $f['class'] : null,
+                    'type' => isset($f['type']) ? $f['type'] : null,
+                    'function' => isset($f['function']) ? $f['function'] : null,
                 ) + $frames[$i - 1],
                 $trace->keepArgs,
                 true
             );
+            if (array($prefix.'src') === array_keys($f = self::castFrameStub($frame, array(), $frame, true))) {
+                $frame = $f[$prefix.'src'];
+                if (1 === count($frame->value)) {
+                    foreach ($frame->value as $label => $frame) {
+                    }
+                } else {
+                    $label = '';
```

Comment:
Can this statement be moved before the `if` condition to prevent the `else` statement?